### DISCUSSION TITLE
Adds a border of the specified width around the image

### DISCUSSION THREAD

**Discussion by Yuj1010** [ImageMagick/ImageMagick#5414] [Help]
hello

I want to add borders of specified width to the top, bottom, left and right of the image.Is there a simple parameter of a CLI option to do that?

thanks

**Comment by fmw42** [comment]
if the same border on all sides, you can use -border. If different on all sides, you will have to use -splice 4 times. See https://github.com/ImageMagick/ImageMagick/discussions/5414

**Comment by GeeMack** [comment]
To add a border of 12 pixels to every side of an image...

    magick logo: -border 12 result.png

To add a border of 12 pixels to just the top and bottom edges...

    ... -border 0x12 ...

To add a border of 12 pixels to only the left and right sides...

    ... -border 12x0 ...

To add a border of 12 pixels to the left and right, *and* a border of 24 pixels to the top and bottom...

    ... -border 12x24 ...

You can also add borders specifying a percentage, like to add 10% of the width to the left and right and 15% of the height to the top and bottom...

    ... -border 10x15% ...

Borders can also be added to individual sides by setting `-gravity` and using `-splice` instead of `-border`.
